zerver/actions/message_edit.py

```diff
@@ -385,6 +385,34 @@ def get_visibility_policy_after_merge(
     return UserTopic.VisibilityPolicy.INHERIT
 
 
+def apply_automatic_unmute_follow_topics_policy(
+    user_profile: UserProfile,
+    message: Message,
+    target_stream: Stream,
+    target_topic_name: str,
+) -> None:
+    if (
+        message.sender.automatically_follow_topics_policy
+        == UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_INITIATION
+    ):
+        bulk_do_set_user_topic_visibility_policy(
+            [message.sender],
+            target_stream,
+            target_topic_name,
+            visibility_policy=UserTopic.VisibilityPolicy.FOLLOWED,
+        )
+    elif (
+        message.sender.automatically_unmute_topics_in_muted_streams_policy
+        == UserProfile.AUTOMATICALLY_CHANGE_VISIBILITY_POLICY_ON_INITIATION
+    ):
+        bulk_do_set_user_topic_visibility_policy(
+            [message.sender],
+            target_stream,
+            target_topic_name,
+            visibility_policy=UserTopic.VisibilityPolicy.UNMUTED,
+        )
```

Comment:
This logic does not seem to match the spec -- there's no check for how the user's existing policy relates to the new one.

Reply:
updated!

Reply:
I don't think we need `select_related` here. Also, we should avoid `.get()` pattern as [we discussed earlier](https://github.com/zulip/zulip/pull/28980#discussion_r1502084484).

This can be simplified to:
```
Subscription.objects.filter(recipient=target_stream.recipient, user_profile=user_profile,
active=True, is_user_active=True)
```

Note: `active=True` is important here, otherwise a user who had unsubscribed the stream will have an entry in UserTopic table -- which is not correct. (Maybe this will raise error in this case -- haven't tested manually).